File: Agora-RTM2-Tutorial-Android-Java/app/src/main/java/io/agora/rtmdemo/EventListener.java
package io.agora.rtmdemo;

import static io.agora.utils.RtmUtils.LogLevel.INFO;

import io.agora.rtm.LinkStateEvent;
import io.agora.rtm.LockEvent;
import io.agora.rtm.MessageEvent;
import io.agora.rtm.PresenceEvent;
import io.agora.rtm.RtmConstants.RtmConnectionState;
import io.agora.rtm.RtmConstants.RtmConnectionChangeReason;
import io.agora.rtm.RtmEventListener;
import io.agora.rtm.StorageEvent;
import io.agora.rtm.TopicEvent;
import io.agora.utils.LogUtil;
import io.agora.utils.RtmUtils.LogLevel;

public class EventListener implements RtmEventListener {
    private LogUtil mLogUtil;

    public EventListener() {
    }

    public void setLogUtil(LogUtil logUtils) {
        mLogUtil = logUtils;
    }

    private void log(LogLevel level, String message) {
        if (mLogUtil != null) {
            mLogUtil.log(level, message);
        }
    }

    @Override
    public void onLinkStateEvent(LinkStateEvent event) { log(INFO, event.toString()); }
    @Override
    public void onMessageEvent(MessageEvent event) {
        log(INFO, event.toString());
    }

    @Override
    public void onPresenceEvent(PresenceEvent event) {
        log(INFO, event.toString());
    }

    @Override
    public void onTopicEvent(TopicEvent event) {
        log(INFO, event.toString());
    }

    @Override
    public void onLockEvent(LockEvent event) {
        log(INFO, event.toString());
    }

    @Override
    public void onStorageEvent(StorageEvent event) {
        log(INFO, event.toString());
    }

    @Override
    public void onConnectionStateChanged(String channelName, RtmConnectionState state, RtmConnectionChangeReason reason) {
        log(INFO, "onConnectionStateChanged, channel name: " + channelName + ", state " + state + ", reason " + reason);
    }

    @Override
    public void onTokenPrivilegeWillExpire(String channelName) {
        log(INFO, "onTokenPrivilegeWillExpire, channel name: " + channelName);
    }
}

File: Agora-RTM2-Tutorial-Android-Kotlin/app/src/main/java/io/agora/rtmdemo/EventListener.kt
package io.agora.rtmdemo

import io.agora.rtm.*
import io.agora.utils.LogUtil
import io.agora.utils.RtmUtils.LogLevel

class EventListener : RtmEventListener {
    var logUtil: LogUtil? = null

    override fun onLinkStateEvent(event: LinkStateEvent) {
        log(LogLevel.INFO, event.toString())
    }

    override fun onMessageEvent(event: MessageEvent) {
        log(LogLevel.INFO, event.toString())
    }

    override fun onPresenceEvent(event: PresenceEvent) {
        log(LogLevel.INFO, event.toString())
    }

    override fun onTopicEvent(event: TopicEvent) {
        log(LogLevel.INFO, event.toString())
    }

    override fun onLockEvent(event: LockEvent) {
        log(LogLevel.INFO, event.toString())
    }

    override fun onStorageEvent(event: StorageEvent) {
        log(LogLevel.INFO, event.toString())
    }

    override fun onConnectionStateChanged(
        channelName: String,
        state: RtmConstants.RtmConnectionState,
        reason: RtmConstants.RtmConnectionChangeReason
    ) {
        log(LogLevel.INFO,"onConnectionStateChanged, channel name: $channelName, state $state, reason $reason")
    }

    override fun onTokenPrivilegeWillExpire(channelName: String) {
        log(LogLevel.INFO,"onTokenPrivilegeWillExpire, channel name: $channelName")
    }

    private fun log(level: LogLevel, message: String) {
        logUtil?.log(level, message)
    }
}